src/java/map/ConcurrentFloatMask.java

```diff
@@ -57,6 +58,18 @@ public ConcurrentFloatMask add(ConcurrentBinaryMask other, float value) {
         );
     }
 
+    public ConcurrentFloatMask subtract(ConcurrentFloatMask other) {
+        return Pipeline.add(this, Arrays.asList(this, other), res ->
+                this.floatMask.add(((ConcurrentFloatMask) res.get(1)).getFloatMask())
+        ).multiply(-1);
+    }
+
+    public ConcurrentFloatMask subtract(ConcurrentBinaryMask other, float value) {
+        return Pipeline.add(this, Arrays.asList(this, other), res ->
+                this.floatMask.add(((ConcurrentBinaryMask) res.get(1)).getBinaryMask(), value)
+        ).multiply(-1);
+    }
+
```

Comment:
This should just use the subtract function in FloatMask so floatmask.subtract rather than duplicating

Reply:
this.floatMask.subtract(((ConcurrentBinaryMask) res.get(1)).getBinaryMask(), value))